packages/matchbox/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,335 @@
+   */
+  outlineBorder?: boolean;
+  variant?: 'outline' | 'mutedOutline' | 'text' | 'filled';
+  size?: 'default' | 'small' | 'large';
+  to?: string;
+  fullWidth?: boolean;
+  submit?: boolean;
+  external?: boolean;
+  title?: string;
+  children?: React.ReactNode;
+}
+
+type PolymorphicButton = Polymorphic.ForwardRefComponent<'button', BaseProps &
+  MarginProps &
+  WidthProps &
+  PaddingProps>;
+
+const Button = React.forwardRef(
+  (props, ref) => {
+    const {
+      children,
+
+      // Styles
+      primary, // Deprecate in favor of color
+      color,
+      disabled,
+      destructive, // Deprecate in favor of color
+      loading,
+      loadingLabel,
+
+      // Below 3 props to be deprecated for a 'variant' prop
+      plain, // Deprecate in favor of flat
+      flat,
+      outline,
+      outlineBorder,
+      variant,
+
+      // Options
+      // Renaming to prevent `width` and `height` pass through
+      // Size is a valid styled-system prop
+      size: buttonSize,
+      fullWidth,
+      submit,
+
+      to,
+      Component, // Deprecate in favor of component
+      component,
+      as,
+      external,
+      title,
+
+      // Events
+      onClick,
+      onFocus,
+      onBlur,
+
+      className = '',
+      ...rest // TODO remove spreading of unknown props
+    } = props;
+
+    const transitionRef = React.useRef();
+
+    const systemProps = pick(rest, system.propNames);
+    const componentProps = omit(rest);
+
+    // Polyfills deprecrated 'Component' prop — use as instead of component
+    const WrapperComponent = as || component || Component;
+
+    // Polyfills to be deprecrated 'primary' and 'destructive' prop
+    const buttonColor = primary ? 'blue' : destructive ? 'red' : color;
+
+    const buttonVariant = React.useMemo(() => {
+      return getVariant({ variant, outline, outlineBorder, plain, flat });
+    }, [variant, outline, outlineBorder, plain, flat, getVariant]);
+
+    const loaderColor = React.useMemo(() => {
+      return getLoaderColor({ variant: buttonVariant, color });
+    }, [buttonVariant, color]);
+
+    const loadingIndicator = React.useMemo(() => {
+      return (
+        <Transition mountOnEnter unmountOnExit in={loading} timeout={0} nodeRef={transitionRef}>
+          {state => (
+            <StyledLoader state={state} ref={transitionRef}>
+              <Spinner
+                // TODO
+                // @ts-ignore
+                color={loaderColor}
+                size="small"
+                label={loadingLabel}
+                rotationOnly
+              />
+            </StyledLoader>
+          )}
+        </Transition>
+      );
+    }, [loading]);
+
+    const sharedProps = {
+      className,
+      disabled: disabled || loading,
+      fullWidth,
+      onClick,
+      onFocus,
+      onBlur,
+      buttonSize,
+      buttonVariant,
+      buttonColor,
+      ref,
+      loading,
+      ...systemProps,
+      ...componentProps,
+    };
```

Comment:
Should we add types for these?

Reply:
These are already typed no?

Reply:
oh yea my bad, not sure what I was thinking 😆